src/main/kotlin/org/opensearch/commons/alerting/model/Comment.kt

```diff
@@ -0,0 +1,159 @@
+package org.opensearch.commons.alerting.model
+
+import org.opensearch.commons.alerting.util.IndexUtils.Companion._ID
+import org.opensearch.commons.alerting.util.instant
+import org.opensearch.commons.alerting.util.optionalTimeField
+import org.opensearch.commons.alerting.util.optionalUserField
+import org.opensearch.commons.alerting.util.optionalUsernameField
+import org.opensearch.commons.authuser.User
+import org.opensearch.core.common.io.stream.StreamInput
+import org.opensearch.core.common.io.stream.StreamOutput
+import org.opensearch.core.common.io.stream.Writeable
+import org.opensearch.core.xcontent.ToXContent
+import org.opensearch.core.xcontent.XContentBuilder
+import org.opensearch.core.xcontent.XContentParser
+import org.opensearch.core.xcontent.XContentParserUtils.ensureExpectedToken
+import java.io.IOException
+import java.time.Instant
+
+data class Comment(
+    val id: String = NO_ID,
+    val entityId: String = NO_ID,
+    val content: String,
+    val createdTime: Instant,
+    val lastUpdatedTime: Instant?,
+    val user: User?
+) : Writeable, ToXContent {
+
+    @Throws(IOException::class)
+    constructor(sin: StreamInput) : this(
+        id = sin.readString(),
+        entityId = sin.readString(),
+        content = sin.readString(),
+        createdTime = sin.readInstant(),
+        lastUpdatedTime = sin.readOptionalInstant(),
+        user = if (sin.readBoolean()) {
```

Comment:
use ternary for better readability

Reply:
Kotlin does not seem to have a ternary operator (https://discuss.kotlinlang.org/t/ternary-operator/2116), and favors directly using if else as it is more idiomatic. I can make it one line to resemble a ternary operator structure more.